pkg/kube/client.go

```diff
@@ -348,96 +448,98 @@ func (c *Client) namespace() string {
 	return v1.NamespaceDefault
 }
 
-// newBuilder returns a new resource builder for structured api objects.
-func (c *Client) newBuilder() *resource.Builder {
-	return c.Factory.NewBuilder().
-		ContinueOnError().
-		NamespaceParam(c.namespace()).
-		DefaultNamespace().
-		Flatten()
-}
-
-// Build validates for Kubernetes objects and returns unstructured infos.
-func (c *Client) Build(reader io.Reader, validate bool) (ResourceList, error) {
-	validationDirective := metav1.FieldValidationIgnore
+func determineFieldValidationDirective(validate bool) FieldValidationDirective {
 	if validate {
-		validationDirective = metav1.FieldValidationStrict
+		return FieldValidationDirectiveStrict
 	}
 
-	schema, err := c.Factory.Validator(validationDirective)
+	return FieldValidationDirectiveIgnore
+}
+
+func buildResourceList(f Factory, namespace string, validationDirective FieldValidationDirective, reader io.Reader, transformRequest resource.RequestTransform) (ResourceList, error) {
+
+	schema, err := f.Validator(string(validationDirective))
 	if err != nil {
 		return nil, err
 	}
-	result, err := c.newBuilder().
+
+	builder := f.NewBuilder().
+		ContinueOnError().
+		NamespaceParam(namespace).
+		DefaultNamespace().
+		Flatten().
 		Unstructured().
 		Schema(schema).
-		Stream(reader, "").
-		Do().Infos()
+		Stream(reader, "")
+	if transformRequest != nil {
+		builder.TransformRequests(transformRequest)
+	}
+	result, err := builder.Do().Infos()
 	return result, scrubValidationError(err)
 }
 
+// Build validates for Kubernetes objects and returns unstructured infos.
+func (c *Client) Build(reader io.Reader, validate bool) (ResourceList, error) {
+	return buildResourceList(
+		c.Factory,
+		c.namespace(),
+		determineFieldValidationDirective(validate),
+		reader,
+		nil)
+}
+
 // BuildTable validates for Kubernetes objects and returns unstructured infos.
 // The returned kind is a Table.
 func (c *Client) BuildTable(reader io.Reader, validate bool) (ResourceList, error) {
-	validationDirective := metav1.FieldValidationIgnore
-	if validate {
-		validationDirective = metav1.FieldValidationStrict
-	}
-
-	schema, err := c.Factory.Validator(validationDirective)
-	if err != nil {
-		return nil, err
-	}
-	result, err := c.newBuilder().
-		Unstructured().
-		Schema(schema).
-		Stream(reader, "").
-		TransformRequests(transformRequests).
-		Do().Infos()
-	return result, scrubValidationError(err)
+	return buildResourceList(
+		c.Factory,
+		c.namespace(),
+		determineFieldValidationDirective(validate),
+		reader,
+		transformRequests)
 }
 
-func (c *Client) update(original, target ResourceList, force, threeWayMerge bool) (*Result, error) {
+func (c *Client) update(target, original ResourceList, updateApplyFunc func(target, original *resource.Info) error) (*Result, error) {
```

Comment:
nit: Was there a reason to swap `target, original`? It makes the review slightly confusing.

Reply:
There was a few different conventions in this file. So I standardized to "target, original". But, err, yeah, I think you're right. I should have standardized to "original, target". https://github.com/helm/helm/pull/31030/commits/a011e08bfd78c705008b310e183a91800b5982d0 does this.